pkg/tools/awarenessgroups/config_decoder_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awarenessgroups
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_AwarenessGroupsDecoder(t *testing.T) {
```

Comment:
Could we add more usecases?



Usecases:



1. empty input
2. list of groups [nsurl1, nsurl2], [nsurl3], [nsurl4]
3. wrong inputs like `[[nsurl]` or `nsurl]]`, or ` [nsurl1, nsurl2], [[nsurl3],, [nsurl4]`


```suggestion
func Test_AwarenessGroupsDecoder(t *testing.T) {
```